Sources/Themes.php

```diff
@@ -920,62 +920,49 @@ function PickTheme()
 		'name' => $txt['theme_pick'],
 	);
 	$context['default_theme_id'] = $modSettings['theme_default'];
-
 	$_SESSION['id_theme'] = 0;
-
-	if (isset($_GET['id']))
-		$_GET['th'] = $_GET['id'];
 	if (!isset($_REQUEST['u']))
 		$_REQUEST['u'] = $user_info['id'];
 
-	// Saving a variant cause JS doesn't work - pretend it did ;)
+	// Have we made a decision, or are we just browsing?
 	if (isset($_POST['save']))
 	{
-		// Which theme?
-		foreach ($_POST['save'] as $k => $v)
-			$_GET['th'] = (int) $k;
+		checkSession();
 
+		$id_theme = (int) key($_POST['save']);
 		if (isset($_POST['vrt'][$k]))
```

Comment:
$k is undefined here so can't change variants, everything else is working as it should now
just gotta make sure that whatever changes made here don't recreate the issue #4527 is meant to fix